compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
Hmm... But then you just need to attach the location of the nested class with an annotation, no? Not sure that is much cleaner. Also it is a step backwards for binary compatibility, no? (Can't change the jsname of a nested class, top-level class is fine though)...